tests/rkt_stop_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+func TestRktStop(t *testing.T) {
+	const imgName = "rkt-stop-test"
+
+	image := patchTestACI(fmt.Sprintf("%s.aci", imgName), fmt.Sprintf("--name=%s", imgName), "--exec=/inspect --read-stdin")
+	defer os.Remove(image)
+
+	imageHash := getHashOrPanic(image)
+	imgID := ImageID{image, imageHash}
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	// Define tests
+	tests := []struct {
+		cmd        string
+		expectKill bool
+	}{
+		// Test regular stop
+		{
+			"stop",
+			false,
+		},
+		// Test forced stop
+		{
+			"stop --force",
+			true,
+		},
+	}
+
+	// Run tests
+	for i, tt := range tests {
+		// Prepare image
+		cmd := fmt.Sprintf("%s --insecure-options=image prepare %s", ctx.Cmd(), imgID.path)
+		podUUID := runRktAndGetUUID(t, cmd)
+
+		// Run image
+		cmd = fmt.Sprintf("%s --insecure-options=image run-prepared --interactive %s", ctx.Cmd(), podUUID)
+		child := spawnOrFail(t, cmd)
+
+		// Sleep to make sure the pod is started
+		time.Sleep(1 * time.Second)
```

Comment:
Since the image is started with `--read-stdin`, it will print "Enter text:".

Instead of using `time.Sleep`, you can use

```
if err := expectTimeoutWithOutput(child, "Enter text:", time.Minute); err != nil {
```

See tests/rkt_interactive_test.go for an example.